image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,137 @@
+---
+title: remove-registry-baremetal
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@dmage"
+  - "@bparees"
+  - "@sdodson"
+  - "@rgolangh" # RHV
+  - "@russellb" # k8s-native infra
+approvers:
+  - "@bparees"
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+creation-date: 2019-09-25
+last-updated: 2019-09-25
+status: provisional
+see-also: 
+replaces:
+superseded-by:
+---
+
+# Remove Registry on Baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+On `baremetal` platform clusters, the OpenShift image registry operator will bootstrap itself as `Removed`.
+This will allow `openshift-installer` to complete installations on this platform type.
+A separate process will be responsible for configuring storage and updating the image registry operator's state to `Managed`.
+
+## Motivation
+
+IPI installation options for Kubernetes-native infrastructure (formerly RHHI.next) and Red Hat Virtualization (RHV) have been declared a key initiative for OpenShift 4.3.
+These will configure themselves as the `baremetal` platform type.
+
+In 4.2 the registry bootstraps itself to use `EmptyDir` storage on `baremetal` to enable pre-GA developemnt on these platforms ([PR #332](https://github.com/openshift/cluster-image-registry-operator/pull/332)).
+This is not suitable for production as there is a serious risk of data loss if a permanent form of storage is not configured.
+
+### Goals
+
+- `openshift-installer` completes if the cloud provider is set to `baremetal`.
+- Image registry is able to bring itself up after the management state is switched from `Removed` to `Managed`.
+
+### Non-Goals
+
+- Provision storage for the registry for RHV and k8s-native infrastructre.
+- Provide 1st class support for additional storage types/providers for the registry (ex: MCG).
+- Harden OpenShift features (ex: Builds, ImageStreams) such that they error cleanly if the registry is not installed.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Story
+
+As an OpenShift cluster admin/installer
+I want the registry to be `Removed` on RHV and k8s-native infra installations
+So that I can provision other storage providers
+And later enable the image registry with the desired storage
+
+### Implementation Details/Notes/Constraints
+
+- When the image registry operator bootstraps itself, it will inspect the cluster infrastructure platform status.
+- If the platform status type is `baremetal`, the registry will mark itself `Removed`.
```

Comment:
`baremetal` or `ovirt`

Reply:
As a reference, see https://github.com/openshift/enhancements/pull/45/files#r329949999

Reply:
@ricardomaraschini scope will be updated to be `baremetal` or `ovirt`

Reply:
@adambkaplan do we need to add VSphere to this proposal too?

Reply:
Reproducing here the conclusions of our last architecture meeting:

1. Platform is known and we have code to support storage backend creation(e.g. AWS): we configure the image registry to use the storage backend and it should be fully functional.
2. Platform is known and it **doesn't** provide storage out of the box(e.g. Baremetal, OVirt): we bootstrap it as _Removed_, configuration is now a day-2 operation.
3. Platform is unknown, we bootstrap with `EmptyDir` to make tests in new platforms easier.

This is the desired state after the implementation:

**AWSPlatformType**  = _Bootstrap in fully working state_
**AzurePlatformType**  = _Bootstrap in fully working state_
**GCPPlatformType**  = _Bootstrap in fully working state_
**OpenStackPlatformType**  = _Bootstrap in fully working state_
**BareMetalPlatformType**  = _Bootstrap as Removed_
**OvirtPlatformType**  = _Bootstrap as Removed_
**LibvirtPlatformType**  = _Bootstrap as Removed_
**NonePlatformType**  = **When do we use this platform?** _Bootstrap as Removed?_
**VSpherePlatformType**  = _Bootstrap as Removed_
**Unknown other platforms** = _Bootstrap using EmptyDir(ephemeral storage)_

/cc @adambkaplan @dmage @bparees